specifications/sep-payment-channel-mechanism.md

```diff
@@ -593,6 +598,36 @@ account has on making a payment, due to liabilities, or would exceed limits on
 the receiving account, such as a trustline limit. Participants must ensure that
 the payments they sign for are receivable by the escrow accounts.
 
+### Trustline Authorization
+
+Any trustlines on the escrow accounts that have been auth revoked, or could be
+auth revoked, could compromise the payment channels ability to close
+successfully.
+
+If the issuer of any auth revocable asset submits an allow trust operation
+freezing the amounts in either escrow account, the close transaction may fail to
+process if its payment operations are dependent on amounts frozen.
+
+There is nothing participants can do to prevent this, other than using only auth
+immutable assets.
+
+### Clawback
+
+Any trustlines on the escrow accounts that have clawback enabled could
+compromise the payment channels ability to close successfully.
+
+If the issuer of any clawback enabled trustline submits a clawback operation for
```

Comment:
Could we treat a clawback as if it were a withdrawal operation?

Reply:
Conceptually they're the same, but a withdrawal operation under normal circumstances requires both parties to sign new C_i and D_i transactions before the withdrawal transaction is signed. The signing of C_i and D_i first is required so that the channel has a valid closing state prepared before the withdrawal changes the state of the channel. There would be no way to guarantee that the participants would know they needed to sign C_i and D_i before a clawback occurred. We could define in a new section what participants should do if a clawback occurs.

However, I think it would be difficult to prescribe what they would do in all cases where clawback occurred. There are are three scenarios I see, assuming clawback of a single asset only:

1. The clawed back amount is an amount that belongs to both parties. I.e. the latest closing transaction at that time makes a payment using the amount. In this case both parties have a loss, but the amount that both parties lose isn't undefined. If EI has $100, $50 owed to ER, and $75 is clawed back, how should the loss be apportioned?

2. The clawed back amount is an amount that belongs to both parties, but the clawed back amount doesn't break the closing transaction. I.e. the latest closing transaction at that time makes a payment using the amount, but the escrow account clawed back from still has sufficient balance of that asset to make the closing transaction payment. This has similar properties are option 1, but at least the channel isn't broken. The closing transaction will still work, but it's possible the issuer knows about the channel and that an amount is owed to the other participant and actually wants to take from them.

3. The clawed back amount is an amount that belongs to a single party. I.e. the latest closing transaction at that time makes no payments using the amount. In this case it is clear only one party is at a loss, and in this case it operates like a withdrawal, assuming all parties see it occur. In this case the channel isn't broken actually, which is nice. However, if one party sees it occur before the other, they may trick the other party into accepting a payment for the amount. Participants are not required to monitor the blockchain constantly. They need only monitor the blockchain once per observation period.

The channel is technically only broken in scenario 1. In scenario 2 the channel can still close, but we don't know if the closure will be fair. In scenario 3 the channel can still close and it will be fair.